bssl-compat/BUILD

```diff
@@ -15,7 +15,12 @@ cmake(
     visibility = ["//visibility:public"],
     generate_crosstool_file = False,
     tags = ["requires-network"],
-    env = { "GOCACHE" : "/tmp" },
+    env = { "GOCACHE" : "/tmp",
+            "CMAKE_C_COMPILER" : "clang",
+            "CMAKE_CXX_COMPILER" : "clang++",
```

Comment:
When invoking cmake, bazel already passes `CC` & `CXX` in the environment, which is enough for cmake to do its thing. Therefore, `CMAKE_C_COMPILER` & `CMAKE_CXX_COMPILER` can be removed.

Reply:
This resulted necessary when building proxy (CMake couldn't find compilers) . However I 'll try again to confirm this.

Reply:
Without `CMAKE_C_COMPILER` or `CMAKE_CXX_COMPILER` in `bssl-compat/BUILD`, I confirmed the following builds still work for me locally:
- Your `dcillera/envoy-openssl[proxy-envoy-openssl]` branch, in the `quay.io/maistra-dev/maistra-builder:2.6` builder.
- Your `dcillera/proxy[for-proxy]` branch, in the upstream builder.